src/client/hud.cpp

```diff
@@ -322,13 +323,34 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				video::SColor color(255, (e->number >> 16) & 0xFF,
 										 (e->number >> 8)  & 0xFF,
 										 (e->number >> 0)  & 0xFF);
-				core::rect<s32> size(0, 0, e->scale.X, text_height * e->scale.Y);
 				std::wstring text = unescape_translate(utf8_to_wide(e->text));
+#ifdef __ANDROID__ // Only for Android!
+				// The text size on Android is not proportional with the actual scaling
+				irr::gui::IGUIFont *font_scaled = g_fontengine->getFont(
+						g_fontengine->getDefaultFontSize() - 3);
+				if (e->offset.X < -20)
+					textsize = font_scaled->getDimension(text.c_str());
+#endif
 				core::dimension2d<u32> textsize = font->getDimension(text.c_str());
 				v2s32 offset((e->align.X - 1.0) * (textsize.Width / 2),
 				             (e->align.Y - 1.0) * (textsize.Height / 2));
-				v2s32 offs(e->offset.X, e->offset.Y);
+#ifdef __ANDROID__ // Android and Desktop
+				core::rect<s32> size(0, 0, e->scale.X * size_factor,
+				                     text_height * e->scale.Y * size_factor);
+				v2s32 offs(e->offset.X * size_factor,
+				           e->offset.Y * size_factor);
+#else
+				core::rect<s32> size(0, 0, e->scale.X, text_height * e->scale.Y);
+				v2s32 offs(e->offset.X, size_factor * e->offset.Y);
+#endif
+#ifdef __ANDROID__ // Only for Android!
+				if (e->offset.X < -20)
+					font_scaled->draw(text.c_str(), size + pos + offset + offs, color);
+				else
+					font->draw(text.c_str(), size + pos + offset + offs, color);
+#else
 				font->draw(text.c_str(), size + pos + offset + offs, color);
+#endif
```

Comment:
Why -20? Why not -30 or -50?

Avoid code duplication.
```suggestion
#ifdef __ANDROID__
				if (e->offset.X < -20) {
					font_scaled->draw(text.c_str(), size + pos + offset + offs, color);
				} else
#endif
				{
					font->draw(text.c_str(), size + pos + offset + offs, color);
				}
```

Reply:
> Why -20? Why not -30 or -50?

I do not care. Trying to avoid unnecessary scaling for text on the side of the screen (e.g. Areas, in-game Time, etc)